sdk/storage/azure-storage-blobs/src/blob_container_client.cpp

```diff
@@ -22,6 +22,78 @@
 
 namespace Azure { namespace Storage { namespace Blobs {
 
+  namespace {
+    Models::BlobItem BlobItemConversion(Models::_detail::BlobItem& item)
+    {
+      Models::BlobItem blobItem;
+      if (item.Name.Encoded)
+      {
+        blobItem.Name = Core::Url::Decode(item.Name.Content);
+      }
+      else
+      {
+        blobItem.Name = std::move(item.Name.Content);
+      }
+      blobItem.IsDeleted = item.IsDeleted;
+      blobItem.Snapshot = std::move(item.Snapshot);
+      blobItem.VersionId = std::move(item.VersionId);
+      blobItem.IsCurrentVersion = item.IsCurrentVersion;
+      blobItem.Details = std::move(item.Details);
+      blobItem.HasVersionsOnly = item.HasVersionsOnly;
+      blobItem.BlobSize = item.BlobSize;
+      blobItem.BlobType = std::move(item.BlobType);
+      if (blobItem.Details.AccessTier.HasValue()
+          && !blobItem.Details.IsAccessTierInferred.HasValue())
+      {
+        blobItem.Details.IsAccessTierInferred = false;
+      }
+      if (blobItem.VersionId.HasValue())
+      {
+        if (!blobItem.HasVersionsOnly.HasValue())
+        {
+          blobItem.HasVersionsOnly = false;
+        }
+        if (!blobItem.IsCurrentVersion.HasValue())
+        {
+          blobItem.IsCurrentVersion = false;
+        }
+      }
+      if (blobItem.BlobType == Models::BlobType::AppendBlob
+          && !blobItem.Details.IsSealed.HasValue())
+      {
+        blobItem.Details.IsSealed = false;
+      }
+      if (blobItem.Details.CopyStatus.HasValue() && !blobItem.Details.IsIncrementalCopy.HasValue())
+      {
+        blobItem.Details.IsIncrementalCopy = false;
+      }
+      {
+        std::map<std::string, std::vector<Models::ObjectReplicationRule>> orPropertiesMap;
+        for (auto& policy : blobItem.Details.ObjectReplicationSourceProperties)
+        {
+          for (auto& rule : policy.Rules)
+          {
+            auto underscorePos = rule.RuleId.find('_', 3);
+            std::string policyId
+                = std::string(rule.RuleId.begin() + 3, rule.RuleId.begin() + underscorePos);
```

Comment:
A comment here describing the format of rule.RuleId might be nice - there's an understanding about the format for this field that isn't obvious (the first 3 characters of the rule id are ignored (why?), then the policy ID is everything in the rule ID before the first underscore character.

That's not at all easy to glean from the code.

Reply:
Added some comments